src/main/java/zd/zero/waifu/motivator/plugin/WaifuMotivatorProject.java

```diff
@@ -39,25 +49,49 @@
     public void projectOpened( @NotNull Project projectOpened ) {
         if ( this.project != null ) return;
 
-        this.project = projectOpened;
-        this.pluginState = WaifuMotivatorPluginState.getPluginState();
-        this.unitTestListener = WaifuUnitTester.newInstance( projectOpened );
-        this.idleEventListener = new IdleEventListener();
+        checkIfInGoodState( projectOpened, () -> {
```

Comment:
The plugin is not usable if any of the `asset.json` files for each asset category are not present.
So it won't register the features unless everything can work offline.